### PR TITLE
[graph] fix top margin location and simplify y-coordinate calculation

### PR DESCRIPTION
**Pull request by midichef**
These changes are for the y-coordinate calculations in InvertedCanvas.

They do not properly take `topMarginPixels` into account, though the problem is hidden because the top margin is currently 0. If we increase it, the new margin shows up below the graph points instead of above, and the y-axis labels become misaligned.

Another problem is that the calculation of y-coordinate inversion is done separately at a few too many interfaces (`plotpixel()`, `scaleY()`, `canvasH()`, `canvasMouse()`), and sometimes with slight differences. That makes it extremely difficult to reason about y coordinates. So I have simplified the inversions, making them go through just one pair of functions: `scaleY()` and its inverse `unscaleY()`. The change happens to fix the `topMarginPixels` bug and simplify `calcTopCursorY()`, which is a sign it's a good way to handle the y coordinates.

I've also added a fix to the y coordinates in `zoom-cursor` (`zz`). The zoom used to go incorrectly to the area above the cursor. Now it zooms into it.

### CODE REVIEW

File: visidata/graph.py [hunk 1]
```diff
@@ -12,35 +12,75 @@ def numericCols(vd, cols):
...
+        return (self.plotviewBox.ymax-plotterY_inverted)/self.yScaler + self.visibleBox.ymin
 
     @property
     def canvasMouse(self):
         p = super().canvasMouse
-        p.y = self.visibleBox.ymin + (self.plotviewBox.ymax-self.plotterMouse.y)/self.yScaler
+        p.y = self.unscaleY(self.plotterMouse.y)
         return p
 
+    def render_sync(self):
```
Comment (saulpw) [inline visidata/graph.py]: I guess this is almost-but-not-exactly identical to Canvas.render_sync?  Is there any reasonable way to parameterize it for the inverted canvas, without duplicating the code?

File: visidata/graph.py [hunk 2]
```diff
@@ -10,6 +10,10 @@ def numericCols(vd, cols):
 
 
 class InvertedCanvas(Canvas):
+    @asyncthread
+    def render_async(self):
+        self.plot_elements(True)
```
Comment (saulpw) [inline visidata/graph.py]: This is all great, but we should probably use `invert_y=True` for good measure.

**Review by saulpw — APPROVED** [review]
Thanks again, this is really helpful.  I wish you'd email me so I can send you some stickers! =)

**Comment by midichef** [comment]
Oops sorry, I have just rebased the squashed version onto the latest develop commit.

**Pull request merged**